modules/schema_validation.lua

```diff
@@ -35,315 +26,264 @@ local function dump(o)
   end
 end
 
-function module.json_validate(table1, table2)
-  local avoid_loops = {}
-  local function recurse(t1, t2)
-    if (type(t1) ~= type(t2)) then return false end
-    if (type(t1) ~= "table") then return t1 == t2 end
-    if (avoid_loops[t1]) then return avoid_loops[t1] == t2 end
-
-    avoid_loops[t1] = t2
-    local t2keys = {}
-    local t2tablekeys = {}
-
-    for k, _ in pairs(t2) do
-      if type(k) == "table" then table.insert(t2tablekeys, k) end
-      t2keys[k] = true
-    end
-
-    for k1, v1 in pairs(t1) do
-      local v2 = t2[k1]
-      if (type(k1) == "table") then
-        local ok = false
-        for i, tk in ipairs(t2tablekeys) do
-          if table_eq(k1, tk) then
-            table.remove(t2tablekeys, i)
-            t2keys[tk] = nil
-            ok = true
-            break
-          end
-        end
-        if not ok
-          then return false,"Missing one or more fields"
-        end
+local function errorMsgToString(tbl)
+  local tmp = ''
+  if type(tbl) == 'table' then
+    for _,v in pairs(tbl) do
+      if type(v) == 'table' then
+        tmp = tmp ..errorMsgToString(v)
       else
-        if (t2keys[k1]) then
-          if not v2 then
-            return false, string.format("Missing value for: '%s'",k1)
-          end
-          t2keys[k1] = nil
-        else
-          t2keys[k1] = v1
-          return false, string.format("Missing: '%s'",k1)
+        if v~="" then
+         tmp = tmp ..v ..'\n'
         end
       end
     end
-
-    if (next(t2keys)) then return false,"Missing one or more fields" end
-    return true
+    return tmp
```

Comment:
Will it be more readable to [reorder](https://adc.luxoft.com/confluence/x/kQTyE) liek following:
```
ifif type(tbl) == 'table' then
  return nil
local tmp;
for ... do 
  ...
```